scripts/skills/perks/perk_rf_fresh_and_furious.nut

```diff
@@ -45,11 +48,31 @@ this.perk_rf_fresh_and_furious <- ::inherit("scripts/skills/skill", {
 
 	function isEnabled()
 	{
-		return this.getContainer().getActor().getFatigue() < 0.3 * this.getContainer().getActor().getFatigueMax();
+		return !this.m.RequiresRecover && this.getContainer().getActor().getFatigue() < this.m.FatigueThreshold * this.getContainer().getActor().getFatigueMax();
 	}
 
 	function onAfterUpdate( _properties )
 	{
+		// We do it like this instead of checking it in onTurnStart directly because some skills e.g. Encumbrance
+		// may modify the fatigue during onTurnStart, so we want to make sure that we check fatigue after that
+		if (this.m.IsStartingTurn)
+		{
+			this.m.IsStartingTurn = false;
+
+			if (this.getContainer().getActor().getFatigue() == 0)
+			{
+				this.m.RequiresRecover = false;
+			}
+			else
+			{
+				if (this.m.RequiresRecover || this.getContainer().getActor().getFatigue() >= this.m.FatigueThreshold * this.getContainer().getActor().getFatigueMax())
+				{
+					this.m.RequiresRecover = true;
```

Comment:
This must be `this.m.RequiresRecover = false;`

Reply:
No it is correct. If the fatigue is greater than 30% then we need to set `RequiresRecover` to true.

Reply:
you're right